corehq/apps/locations/models.py

```diff
@@ -303,8 +303,26 @@ class LocationQuerySet(LocationQueriesMixin, CTEQuerySet):
     pass
 
 
+def location_queryset(func):
+    @wraps(func)
+    def wrapper(self, *args, **kw):
+        result = func(self, *args, **kw)
+        if type(result) == CTEQuerySet:
+            result.__class__ = LocationQuerySet
```

Comment:
Yikes, this feels scary.  Do you expect this to go away after we kill MPTT?

Reply:
This is just making sure the returned queryset has methods like `location_ids` because django-cte's `With.queryset()` returns a generic `CTEQuerySet`. I'll see what I can do to make it return the same type of queryset that would be returned by the model manager.

Reply:
https://github.com/dimagi/commcare-hq/pull/20153